### PR TITLE
[next] Hide prompt input for secure properties

### PR DESCRIPTION
**Pull request by t1m0thyj**
Part of https://github.com/zowe/zowe-cli/issues/1106, but does not completely resolve it

* Detect if profile properties are designated as secure in team config. If so, then add `(will be hidden)` to the prompt message (like the Vault CLI does) and mask the prompt input.
* Fix bugs that I found in ConfigAutoStore when testing. It failed to store properties in cases where either the service or base profile was missing from team config.
* Move the `setupConfigToLoad` method used by several test suites into our TestUtil module to reduce code duplication.

### CODE REVIEW

**Review by gejohnston — APPROVED** [review]
Looks good.

**Review by awharn — COMMENTED** [review]
I have one question before I approve.

File: packages/config/src/ConfigAutoStore.ts [hunk 1]
```diff
@@ -125,10 +148,16 @@ export class ConfigAutoStore {
...
-            // Determine if property should be stored in base profile instead
-            if (loadedProfile.properties[propName] == null && !loadedProfile.secure?.includes(propName) &&
-                (baseProfileObj.properties[propName] != null || baseProfileObj.secure?.includes(propName) ||
-                (propName === "tokenValue" && baseProfileObj.properties.tokenType != null))) {
+            /* If any of the following is true, then property should be stored in base profile:
+                (1) Service profile does not exist, but base profile does
+                (2) Property is missing from service profile properties/secure objects, but present in base profile
+                (3) Property is tokenValue and tokenType is missing from service profile, but present in base profile
+            */
+            if ((loadedProfile == null && baseProfileObj != null) ||
```
[the remark this answers is not in the document]
Comment (t1m0thyj) [reply packages/config/src/ConfigAutoStore.ts:156]: Yes, I tested the following scenarios with a nested config (will send internally) and they all worked to auto-store credentials:
* `secure` array includes user and password in service profile
* `secure` array includes user and password in base profile
* `secure` arrays are empty so auto-store picks service profile

Would you suggest to add an automated test that verifies auto-store works for a nested config? 😛

[the remark this answers is not in the document]
Comment (t1m0thyj) [reply packages/config/src/ConfigAutoStore.ts:156]: Found a problem with nested config 😢 If the `secure` array is defined at a higher level, profiles may not inherit it correctly.

**Review by gejohnston — APPROVED** [review]

File: packages/config/src/ConfigAutoStore.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (t1m0thyj) [reply packages/config/src/ConfigAutoStore.ts]: Fixed this problem and added unit test to verify that auto-store works for a nested config. Marked PR as ready for review again 🙂

**Review by awharn — APPROVED** [review]

**Pull request merged**